plugins/pastefromwordimage/plugin.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or http://ckeditor.com/license
+ */
+
+( function() {
+	'use strict';
+
+	CKEDITOR.plugins.add( 'pastefromwordimage', {
+		requires: 'pastefromword',
+		init: function() {}
+	} );
+
+	/**
+	 * Help methods used by paste from word image plugin.
+	 *
+	 * @since 4.8.0
+	 * @class CKEDITOR.plugins.pastefromwordimage
+	 */
+	CKEDITOR.plugins.pastefromwordimage = {
+		/**
+		 * Methods parse rtf clipboard to find embeded images.
+		 *
+		 * @private
+		 * @since 4.8.0
+		 * @param {String} rtfClipboard Data obtained from rtf clipboard.
+		 * @returns {Array} Contains array of objects with images.
+		 * @returns {Object} return.Object Single image found in `rtfClipboard`.
+		 * @returns {String/null} return.Object.hex Hexadecimal string of image embeded in rtf clipboard.
+		 * @returns {String/null} return.Object.type String represent type of image, allowed values: 'image/png', 'image/jpeg' or `null`
+		 */
+		extractImagesFromRtf: function( rtfClipboard ) {
+			var images = [],
+				rePictureOrShape = /((?:\{\\\*\\shppict[\s\S]+?{\\\*\\blipuid\s+[0-9a-f]+\}\s?)|(?:\{\\shp[\s\S]+?\{\\\*\\svb\s?))([0-9a-f\s]+)\}\}/g,
+				rePictureHeader = /\{\\\*\\shppict[\s\S]+?{\\\*\\blipuid\s+[0-9a-f]+\}\s?/,
+				reShapeHeader = /\{\\shp[\s\S]+?\{\\\*\\svb\s?/,
+				wholeImage,
+				imageType;
+
+			wholeImage = rtfClipboard.match( rePictureOrShape );
+			if ( !wholeImage ) {
+				return;
+			}
+
+			for ( var i = 0, len = wholeImage.length; i < len; i++ ) {
```

Comment:
There's very little if any benefit from extracting len into a var, while it decreases the readability.